applications/GeoMechanicsApplication/custom_elements/U_Pw_small_strain_element.cpp

```diff
@@ -640,13 +622,10 @@ void UPwSmallStrainElement<TDim, TNumNodes>::CalculateOnIntegrationPoints(
             CalculateFluidFluxes(GeoTransportEquationUtilities::CalculatePermeabilityUpdateFactors(
                                      strain_vectors, this->GetProperties()),
                                  rCurrentProcessInfo);
-        for (unsigned int GPoint = 0; GPoint < NumGPoints; ++GPoint) {
+        for (unsigned int GPoint = 0; GPoint < number_of_integration_points; ++GPoint) {
             GeoElementUtilities::FillArray1dOutput(rOutput[GPoint], fluid_fluxes[GPoint]);
         }
     } else {
-        if (rOutput.size() != mConstitutiveLawVector.size())
-            rOutput.resize(mConstitutiveLawVector.size());
-
         for (unsigned int i = 0; i < mConstitutiveLawVector.size(); ++i) {
             noalias(rOutput[i]) = ZeroVector(3);
```

Comment:
The reason I didn't do that is `rOutput[i]` is provided as an in/out argument of the call to `GetValue`. From the caller's point of view, I have no clue as to what this function expects. To avoid accidentally changing the current behavior, I thought it would be better to leave the initial assignment in place. I do agree, however, that it is better to get rid of the `noalias` part in the assignment, so I have done that.